src/Symfony/Component/ErrorHandler/DebugClassLoader.php

```diff
@@ -849,6 +849,26 @@ private function setReturnType(string $types, string $class, string $method, str
         $docTypes = [];
 
         foreach ($typesMap as $n => $t) {
+            if (str_contains($n, '::')) {
+                [$definingClass, $constantName] = explode('::', $n, 2);
+                $definingClass = match ($definingClass) {
+                    'self', 'static', 'parent' => $class,
+                    default => $definingClass,
+                };
+
+                if (!\defined($definingClass.'::'.$definingClass)) {
+                    return;
+                }
+
+                $constant = new \ReflectionClassConstant($definingClass, $constantName);
+
+                if (method_exists(\ReflectionClassConstant::class, 'getType') && $constantType = $constant->getType()) {
```

Comment:
Can we replace the `method_exists` check with a `PHP_VERSION_ID` check ? This has 2 benefits:

- it will make it clearer when we can simplify that check thanks to our min requirement
- it will be resolved at compile-time

Reply:
done